src/array_api_stubs/_draft/elementwise_functions.py

```diff
@@ -836,10 +836,12 @@ def clip(
     Notes
     -----
 
+    - This function is conceptually equivalent to ``maximum(minimum(x, max), min)`` when ``x``, ``min``, and ``max`` have the same data type.
+    - For scalar ``min`` and/or ``max``, the scalar values must be converted to zero-dimensional arrays having the same data type as ``x`` prior to broadcasting.
```

Comment:
This sounds a bit like an implementation requirement, rather than only clarifying the intended semantics. It's also not what implementations do for mixed-kind dtypes:
```python
>>> np.clip(np.arange(3, dtype=np.int32), -1, 1.5).dtype
dtype('float64')
```

I suggest saying instead that for scalars regular type promotion rules have to be followed (and hence mixed-kind is undefined).

Reply:
Yeah, but wouldn't that be functionally the same? According to type promotion rules involving scalars and arrays, the first step is to convert the scalar to the same dtype as the array. When I included the line about conversion, that was what I had in mind.

Reply:
Okay. I've updated the text to hopefully address your concerns regarding outside-the-standard behavior.

Reply:
...and I needed to relax the output array data type requirement.